From 05a36ccdfca01c1c51cce8c5913188de7302e547 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@st.com>
Date: Tue, 29 Sep 2020 13:24:27 +0200
Subject: [PATCH] ARM: dts: stm32: add USB OTG wake up support on stm32mp151

Add USB OTG wake up support on stm32mp151:
-use exti 44 line to wakeup from CStop
-attach usbotg device to "pd_core" power domain

Change-Id: I6ca5ac8ef6f677f4534ada05baf1c28505b92b0e
Signed-off-by: Amelie Delaunay <amelie.delaunay@st.com>
---
 arch/arm/boot/dts/st/stm32mp151.dtsi | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/st/stm32mp151.dtsi b/arch/arm/boot/dts/st/stm32mp151.dtsi
index bb72d0e15cbd..e068121c7bf4 100644
--- a/arch/arm/boot/dts/st/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp151.dtsi
@@ -1177,13 +1177,15 @@ usbotg_hs: usb-otg@49000000 {
 			clock-names = "otg", "utmi";
 			resets = <&rcc USBO_R>;
 			reset-names = "dwc2";
-			interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti 44 IRQ_TYPE_LEVEL_HIGH>;
 			g-rx-fifo-size = <512>;
 			g-np-tx-fifo-size = <32>;
 			g-tx-fifo-size = <256 16 16 16 16 16 16 16>;
 			dr_mode = "otg";
 			otg-rev = <0x200>;
 			usb33d-supply = <&usb33>;
+			power-domains = <&pd_core>;
+			wakeup-source;
 			status = "disabled";
 		};
 
-- 
2.39.5

